### DISCUSSION TITLE
How to keep generic type of paramters?

### DISCUSSION THREAD

**Discussion by zjns** [square/retrofit#4173] [Q&A]
What kind of issue is this?

 - [x] Question. This issue tracker is not the place for questions. If you want to ask how to do
       something, or to understand why something isn't working the way you expect it to, use Stack
       Overflow. https://stackoverflow.com/questions/tagged/retrofit

As title say, I want to keep generic type of paramters, but I don't know how to do it, default proguard rules of Retrofit not working.
Can anyone tell me how to do this?
![image_2024-06-19_12-45-07](https://github.com/square/retrofit/assets/10793522/ff3a9b65-e915-40b1-aab6-a292dcbc247a)

**Comment by JakeWharton** [comment]
It is impossible to target that with static shrinker rules. As a result, we offer https://github.com/square/retrofit/tree/trunk/retrofit-response-type-keeper

**Comment by zjns** [comment]
Thanks for the answer, I solved this roughly by turning off R8 fullMode.
